src/AspNetCore/Src/aspnetcoreapplication.cxx

```diff
@@ -0,0 +1,516 @@
+
+
+VOID
+ASPNETCORE_APPLICATION::SetCallbackHandles(
+    _In_ PFN_REQUEST_HANDLER request_handler,
+    _In_ PFN_SHUTDOWN_HANDLER shutdown_handler,
+    _In_ VOID* pvRequstHandlerContext,
+    _In_ VOID* pvShutdownHandlerContext
+)
+{
+    m_RequestHandler = request_handler;
+    m_RequstHandlerContext = pvRequstHandlerContext;
+    m_ShutdownHandler = shutdown_handler;
+    m_ShutdownHandlerContext = pvShutdownHandlerContext;
+
+    // Initialization complete
+    SetEvent(m_InitalizeEvent);
+}
+
+
+
+HRESULT
+ASPNETCORE_APPLICATION::Initialize(
+    _In_ ASPNETCORE_CONFIG * pConfig
+)
+{
+    DWORD dwTimeout;
+    DWORD dwResult;
+
+    m_pConfiguration = pConfig;
+
+    m_InitalizeEvent = CreateEvent(
+        NULL,   // default security attributes
+        TRUE,   // manual reset event
+        FALSE,  // not set
+        NULL);  // name
+
+    if (m_InitalizeEvent == NULL)
+    {
+        return HRESULT_FROM_WIN32(GetLastError());
+    }
+
+    m_hThread = CreateThread(
+        NULL,       // default security attributes
+        0,          // default stack size
+        (LPTHREAD_START_ROUTINE)ExecuteAspNetCoreProcess,
+        this,       // thread function arguments
+        0,          // default creation flags
+        NULL);      // receive thread identifier
+
+    if (m_hThread == NULL)
+    {
+        return HRESULT_FROM_WIN32(GetLastError());
+    }
+
+    // If the debugger is attached, never timeout
+    if (IsDebuggerPresent())
+    {
+        dwTimeout = INFINITE;
+    }
+    else
+    {
+        dwTimeout = pConfig->QueryStartupTimeLimitInMS();
+    }
+
+    // What should the timeout be? (This is sorta hacky)
+    const HANDLE pHandles[2]{ m_hThread, m_InitalizeEvent };
+
+    // Wait on either the thread to complete or the event to be set
+    dwResult = WaitForMultipleObjects(2, pHandles, FALSE, dwTimeout);
+
+    // It all timed out
+    if (dwResult == WAIT_TIMEOUT)
+    {
+        return HRESULT_FROM_WIN32(dwResult);
+    }
+
+    dwResult = WaitForSingleObject(m_hThread, 0);
+    // The thread ended it means that something failed
+    if (dwResult == WAIT_OBJECT_0)
+    {
+        return HRESULT_FROM_WIN32(dwResult);
+    }
+
+    return S_OK;
+}
+
+VOID
+ASPNETCORE_APPLICATION::ExecuteApplication(
+    VOID
+)
+{
+    HRESULT     hr = S_OK;
+
+    STRU            strFullPath;
+    STRU            strDotnetExeLocation;
+    STRU            strHostFxrSearchExpression;
+    STRU            strDotnetFolderLocation;
+    STRU            strHighestDotnetVersion;
+    STRU            strApplicationFullPath;
+    PWSTR           strDelimeterContext = NULL;
+    PCWSTR          pszDotnetExeLocation = NULL;
+    PCWSTR          pszDotnetExeString(L"dotnet.exe");
+    DWORD           dwCopyLength;
+    HMODULE         hModule;
+    PCWSTR          argv[2];
+    hostfxr_main_fn pProc;
+
+    // Get the System PATH value.
+    hr = GetEnv(L"PATH", &strFullPath);
+    if (FAILED(hr))
```

Comment:
what would be the http response if ExecuteApplication failed in the middle?

Reply:
500 server error?